src/vcf/meta_entry.cpp

```diff
@@ -216,16 +219,28 @@ namespace ebi
             type_field != "String") {
             throw new MetaSectionError{entry.line, "INFO metadata Type is not a Integer, Float, Flag, Character or String"};
         }
+        
+        std::map<std::string, std::pair<std::string, std::string>> version_info_type_map;
+        std::map<std::string, std::pair<std::string, std::string>> version_info_number_map;
 
-        for (const auto & t : info_type) {
+        if (entry.source->version == Version::v41 || entry.source->version == Version::v42) {
+            version_info_type_map = info_type_v41_v42;
+            version_info_number_map = info_number_v41_v42;
+        }
+        else if (entry.source->version == Version::v43) {
+            version_info_type_map = info_type_v43;
+            version_info_number_map = info_number_v43;
+        }
+        
+        for (const auto & t : version_info_type_map) {
             if (value["ID"] == t.first) {
                 if (type_field != t.second.first) {
-                    throw new MetaSectionError{entry.line, t.second.second};
+                        throw new MetaSectionError{entry.line, t.second.second};
                 }
             }
         }
 
-        for (const auto & n : info_number) {
+        for (const auto & n : version_info_number_map) {
```

Comment:
when this part is extracted to another function, change variable names to `number` and `info_numbers`. (and `type` and `info_types` in the previous `for`)